File: app/src/main/java/com/github/pockethub/android/ui/item/commit/CommitParentItem.java
package com.github.pockethub.android.ui.item.commit;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.github.pockethub.android.R;
import com.github.pockethub.android.core.commit.CommitUtils;
import com.github.pockethub.android.ui.StyledText;
import com.github.pockethub.android.ui.item.BaseDataItem;
import com.github.pockethub.android.ui.item.BaseViewHolder;
import com.meisolsson.githubsdk.model.Commit;

import butterknife.BindView;

public class CommitParentItem extends BaseDataItem<Commit, CommitParentItem.ViewHolder> {

    private final Context context;

    public CommitParentItem(Context context, Commit commit) {
        super(null, commit, commit.sha().hashCode());
        this.context = context;
    }

    @Override
    public void bind(@NonNull ViewHolder viewHolder, int position) {
        StyledText parentText = new StyledText()
                .append(context.getString(R.string.parent_prefix))
                .monospace(CommitUtils.abbreviate(getData().sha()))
                .underlineAll();
        viewHolder.commitId.setText(parentText);
    }

    @Override
    public int getLayout() {
        return R.layout.commit_parent_item;
    }

    @NonNull
    @Override
    public ViewHolder createViewHolder(@NonNull View itemView) {
        return new ViewHolder(itemView);
    }

    public class ViewHolder extends BaseViewHolder {

        @BindView(R.id.tv_commit_id) TextView commitId;

        public ViewHolder(@NonNull View rootView) {
            super(rootView);
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/commit/CommitParentItem.kt
package com.github.pockethub.android.ui.item.commit

import android.content.Context
import androidx.core.text.buildSpannedString
import androidx.core.text.underline
import com.github.pockethub.android.R
import com.github.pockethub.android.util.android.text.monospace
import com.github.pockethub.android.core.commit.CommitUtils
import com.meisolsson.githubsdk.model.Commit
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.commit_parent_item.*

class CommitParentItem(
        private val context: Context,
        val commit: Commit
) : Item(commit.sha()!!.hashCode().toLong()) {

    override fun getLayout() = R.layout.commit_parent_item

    override fun bind(holder: ViewHolder, position: Int) {
        holder.tv_commit_id.text = buildSpannedString {
            underline {
                append(context.getString(R.string.parent_prefix))
                monospace {
                    append(CommitUtils.abbreviate(commit.sha()))
                }
            }
        }
    }
}
